### PR TITLE
Add GetName function in Symbol class for cpp package

### PR DESCRIPTION
**Pull request by haojin2**
## Description ##
As title.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Add GetName function for Symbol class

## Comments ##
Address #12073

### CODE REVIEW

File: cpp-package/include/mxnet-cpp/symbol.hpp [hunk 1]
```diff
@@ -172,6 +172,13 @@ inline std::vector<std::string> Symbol::ListAuxiliaryStates() const {
   return ret;
 }
 
+inline std::string Symbol::GetName() const {
+  int success;
+  const char* out_name;
+  CHECK_EQ(MXSymbolGetName(GetHandle(), &out_name, &success), 0);
```
Comment (apeforest) [inline cpp-package/include/mxnet-cpp/symbol.hpp]: What if success == 0? Based on the function description, it seems that out_name could be NULL even if the returned value is 0:

```
/*!
 * \brief Get string attribute from symbol
 * \param symbol the source symbol
 * \param key The key of the symbol.
 * \param out The result attribute, can be NULL if the attribute do not exist.
 * \param success Whether the result is contained in out.
 * \return 0 when success, -1 when failure happens
 */
int NNSymbolGetAttr(SymbolHandle in,
                             const char* key,
                             const char** out,
                             int *out);
```

Comment (haojin2) [reply cpp-package/include/mxnet-cpp/symbol.hpp]: added.

**Comment by haojin2** [comment]
@nswamy Should be good for merge? BTW this is a change to cpp-package, I don't think it's related to backend.

**Comment by nswamy** [comment]
Thanks @haojin2 . I changed the label and merging.

**Pull request merged**